File: 1_python_library/fundamentals/02_control_flow.py
#!/usr/bin/env python3
"""Structures de contrôle — parcourir ticks, signaux, positions."""

prices = [100.0, 100.5, 99.8, 100.2, 101.0]
position = 0
threshold = 100.3

# Boucle : générer un signal simple de cassure
signals: list[str] = []
for p in prices:
    if p > threshold and position == 0:
        signals.append("BUY")
        position = 1
    elif p < threshold and position == 1:
        signals.append("SELL")
        position = 0
    else:
        signals.append("HOLD")

print("Prix   :", prices)
print("Signaux:", signals)

# Compréhension : filtrer les ticks « haussiers »
up_ticks = [p for p in prices if p >= 100.0]
print("Ticks >= 100 :", up_ticks)

File: 3_rust_library/fundamentals/examples/02_control_flow.rs
//! Contrôle de flux — signaux sur série de prix

fn signal(price: f64, threshold: f64, long: bool) -> &'static str {
    match (long, price > threshold) {
        (true, true) => "BUY",
        (false, true) => "SELL",
        _ => "HOLD",
    }
}

fn main() {
    let prices = [100.0, 100.5, 99.8, 100.2, 101.0];
    let threshold = 100.3;
    let mut long = false;

    for p in prices {
        let s = signal(*p, threshold, !long);
        if s == "BUY" {
            long = true;
        }
        if s == "SELL" {
            long = false;
        }
        println!("prix={p:.2} -> {s}");
    }
}
